jerry-core/parser/js/js-parser-expr.c

```diff
@@ -1524,6 +1524,8 @@ parser_process_unary_expression (parser_context_t *context_p) /**< context */
                                        CBC_PUSH_IDENT_REFERENCE,
                                        context_p->last_cbc.third_literal_index);
             }
+
+            parser_emit_cbc_ext (context_p, CBC_EXT_RESOLVE_BASE);
```

Comment:
Question: this is always after a CBC_PUSH_IDENT_REFERENCE. Would it be a good idea to do both things in this opcode: push ident ref and base?

Reply:
Wouldn't this involve too many code duplication?

If we wan to do it in one instruction, it'd be look like this:

```c
case VM_OC_PUSH_IDENT_AND_RESOLVE_BASE:
{
  uint16_t literal_index;

  READ_LITERAL_INDEX (literal_index);

  JERRY_ASSERT (literal_index < ident_end);

  if (literal_index < register_end)
  {
    *stack_top_p++ = ECMA_VALUE_UNDEFINED;
    *stack_top_p++ = ECMA_VALUE_UNDEFINED; // Okay, here we can save some instructoins
    *stack_top_p++ = ecma_fast_copy_value (frame_ctx_p->registers_p[literal_index]);
  }
  else
  {
    ecma_string_t *name_p = ecma_get_string_from_value (literal_start_p[literal_index]);

    ecma_object_t *ref_base_lex_env_p;
    ref_base_lex_env_p = ecma_op_resolve_reference_base (frame_ctx_p->lex_env_p,
                                                         name_p);

    result = ecma_op_get_value_lex_env_base (ref_base_lex_env_p,
                                             name_p,
                                             is_strict);

    if (ECMA_IS_VALUE_ERROR (result))
    {
      goto error;
    }

    ecma_ref_object (ref_base_lex_env_p);
    ecma_ref_ecma_string (name_p);
    *stack_top_p++ = ecma_make_object_value (ref_base_lex_env_p);
    *stack_top_p++ = ecma_make_string_value (name_p);
    ecma_value_t this_value = result;
    if (vm_get_implicit_this_value (&this_value))
    {
      ecma_free_value (result);
      result = this_value;
    }
    *stack_top_p++ = result;
  }

   continue;
}
```
I think it isn't worth it. But do you have ideai for it?

Reply:
I am thinking about having a single implementation for the two opcodes:

```
case PUSH_IDENT_SOMETHING:
{
  bool is_resolve_base = byte code is ext opcode;

  if (literal_index < register_end) {
    if (is_resolve_base)
    {
      // Do this
    }
    else
    {
      // Do that
    }
  }
}
```

Reply:
Ohh, I see now! I'll try your suggestion.